src/server.ts

```diff
@@ -1,84 +1,462 @@
-  const createdSignature = addDbItem(postData);
-  res.status(201).json(createdSignature);
+//------------------------------------------------gets top 5 resources
+app.get("/resources/top", async (req, res) => {
+  try {
+    const queryResponse = await client.query(`
+    SELECT * 
+    FROM resources
+    ORDER BY likes DESC
+    LIMIT 5
+    `);
+    const allResources = queryResponse.rows;
+    res.status(200).json(allResources);
+  } catch (error) {
+    console.error(error);
+    res.status(404).json({ message: "error, get top resources" });
+  }
 });
 
-// GET /items/:id
-app.get<{ id: string }>("/items/:id", (req, res) => {
-  const matchingSignature = getDbItemById(parseInt(req.params.id));
-  if (matchingSignature === "not found") {
-    res.status(404).json(matchingSignature);
-  } else {
-    res.status(200).json(matchingSignature);
+//------------------------------------------------get 5 most recent resources
+app.get("/resources/recent", async (req, res) => {
+  try {
+    const queryResponse = await client.query(`
+    SELECT * 
+    FROM resources
+    ORDER BY post_date DESC
+    LIMIT 5
+    `);
+    const allResources = queryResponse.rows;
+    res.status(200).json(allResources);
+  } catch (error) {
+    console.error(error);
+    res.status(404).json({ message: "error, get recent resources" });
   }
 });
 
-// DELETE /items/:id
-app.delete<{ id: string }>("/items/:id", (req, res) => {
-  const matchingSignature = getDbItemById(parseInt(req.params.id));
-  if (matchingSignature === "not found") {
-    res.status(404).json(matchingSignature);
-  } else {
-    res.status(200).json(matchingSignature);
+//------------------------------------------------gets a specific resource if it matches the link
+app.get<{}, {}, {}, { link: string }>("/resources/link", async (req, res) => {
+  try {
+    const queryLink = [req.query.link];
+    const queryResponse = await client.query(
+      `
+    SELECT * 
+    FROM resources
+    WHERE link = $1
+    `,
+      queryLink
+    );
+    const matchingResource = queryResponse.rows;
+    res.status(200).json(matchingResource);
+  } catch (error) {
+    console.error(error);
+    res.status(404).json({ message: "error, get a specific resource" });
   }
 });
 
-// PATCH /items/:id
-app.patch<{ id: string }, {}, Partial<DbItem>>("/items/:id", (req, res) => {
-  const matchingSignature = updateDbItemById(parseInt(req.params.id), req.body);
-  if (matchingSignature === "not found") {
-    res.status(404).json(matchingSignature);
-  } else {
-    res.status(200).json(matchingSignature);
+//------------------------------------------------gets likes given a user_id
+app.get<{ userid: string }>("/likes/:userid", async (req, res) => {
+  try {
+    const queryValues = [req.params.userid];
+    const queryResponse = await client.query(
+      `
+    SELECT *
+    FROM likes
+    WHERE user_id = $1
+    `,
+      queryValues
+    );
+    const allLikeReactions = queryResponse.rows;
```

Comment:
Variable name could be improved, not sure what allLikeReactions means. Did you mean allLikedReactions or GetAllLikedReactions?